### PR TITLE
Added definitions for v1.3.5 of the socket.io lib

### PR DESCRIPTION
**Pull request by divillysausages**
I've moved the old definitions to legacy/socket.io-1.2.0 etc. The tests are the same, as the actual implementation hasn't changed.

When running the tests for the server lib, I get the following errors:

```
../node/node.d.ts(260,26): error TS2304: Cannot find name 'DataView'.
../node/node.d.ts(274,21): error TS2304: Cannot find name 'Map'.
../node/node.d.ts(283,21): error TS2304: Cannot find name 'Set'.
../node/node.d.ts(293,25): error TS2304: Cannot find name 'WeakMap'.
```

Which I guess is linked to Issue #4249? In any case, the errors with the node definitions, not the socket.io ones.

### CODE REVIEW

**Comment by divillysausages** [comment]
Hm, the two errors in Travis are:

```
socket.io/socket.io.d.ts(762,51): error TS1005: ';' expected.
socket.io/socket.io.d.ts(762,70): error TS1005: ';' expected.
```

but the line in question is:

```
broadcast( packet: any, opts: { rooms?: string[], except?: string[], flags?: {[flag: string]: boolean} } ):void;
```

51 and 70 are just after the "t" in "string" - am I missing something?

File: socket.io/socket.io.d.ts [hunk 1]
```diff
@@ -1,96 +1,823 @@
...
+		
+		/**
+		 * Broadcasts a packet
+		 * @param packet The packet to broadcast
+		 * @param opts Any options to send along:
+		 * 	- rooms: An optional list of rooms to broadcast to. If empty, the packet is broadcast to all sockets
+		 * 	- except: A list of Socket IDs to exclude
+		 * 	- flags: Any flags that we want to send along ('json', 'volatile', 'broadcast')
+		 */
+		broadcast( packet: any, opts: { rooms?: string[], except?: string[], flags?: {[flag: string]: boolean} } ):void;
```
Comment (vvakame) [inline socket.io/socket.io.d.ts]: you should use `broadcast( packet: any, opts: { rooms?: string[]; except?: string[]; flags?: {[flag: string]: boolean;}; } ):void;`. not comma.

**Comment by divillysausages** [comment]
I fixed the problem with the broadcast() method, though I guess someone edited the socket.io-client file in the meantime. There was a merge conflict with my changes, but my changes encompass the edit, so everything should be fine

**Comment by divillysausages** [comment]
To be more precise, there was a merge conflict when I updated my forked repo, but the edits I've made should be fine

**Comment by vvakame** [comment]
*socket.io-client/socket.io-client.d.ts*

to author(@divillysausages). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*socket.io/socket.io.d.ts*

to author(@divillysausages). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

**Comment by vvakame** [comment]
please fix travis ci.

**Comment by divillysausages** [comment]
Finally :) Sorry about that; I'm a bit new to pull requests and travis. 

On the contributions page it says: "If you need many attempts to fix errors then please flatten the history and clean the commit message." - need me to do this? Also, how do I do it?

File: socket.io-client/socket.io-client.d.ts [hunk 2]
```diff
@@ -1,46 +1,656 @@
-// Type definitions for socket.io-client 1.2.0
+// Type definitions for socket.io-client 1.3.5
 // Project: http://socket.io/
-// Definitions by: PROGRE <https://github.com/progre/>
+// Definitions by: divillysausages <https://github.com/divillysausages/>
```
Comment (vvakame) [inline socket.io-client/socket.io-client.d.ts]: please keep original author.

**Comment by vvakame** [comment]
@divillysausages np.
@progre could you review this PR?

File: socket.io-client/socket.io-client.d.ts (hunk 2 above)
Comment (divillysausages) [inline socket.io-client/socket.io-client.d.ts]: OK, sorry, I thought the author was updated with each revision. Want me to change it or leave it for @progre when he reviews the PR?

**Comment by progre** [comment]
@vvakame LGTM

File: socket.io-client/socket.io-client.d.ts (hunk 2 above)
Comment (vvakame) [inline socket.io-client/socket.io-client.d.ts]: yes.

Comment (divillysausages) [inline socket.io-client/socket.io-client.d.ts]: Just to clarify - we keep the original author, even if the definitions are for a different version of the lib?

I'll revert the line when I get back to my computer tonight. Tks

Comment (vvakame) [inline socket.io-client/socket.io-client.d.ts]: yep. author is reviewer and 1st class user.
please add your name below progre's name.
like this. https://github.com/borisyankov/DefinitelyTyped/blob/master/jquery/jquery.d.ts#L3

Comment (divillysausages) [inline socket.io-client/socket.io-client.d.ts]: Ah, ok, will do, thanks!

Comment (divillysausages) [inline socket.io-client/socket.io-client.d.ts]: Fixed!

**Comment by vvakame** [comment]
@progre thanks!
@divillysausages thanks mate!

**Pull request merged**